consensus/phases/firstphase.go

```diff
@@ -169,6 +170,39 @@ func detectSparseBitsetLength(claims map[core.RecordRef][]packets.ReferendumClai
 	return 0, errors.New("not implemented")
 }
 
+func (fp *FirstPhase) getSignedClaims(claims []packets.ReferendumClaim) []packets.ReferendumClaim {
+	result := make([]packets.ReferendumClaim, 0)
+	for _, claim := range claims {
+		joinClaim, ok := claim.(*packets.NodeJoinClaim)
+		if ok {
+			signConfirmed, err := fp.claimSignIsOk(joinClaim)
+			if err != nil {
+				log.Error("[ getSignedClaims ] failed to check a claim sign")
+				continue
+			}
+			if !signConfirmed {
+				log.Error("[ getSginedClaims ] sign is unconfirmed")
+				continue
+			}
+		}
+		result = append(result, claim)
```

Comment:
claim will be appended to list despite the failed sign check; is this the desired behaviour?

Reply:
if sign check is failed this iteration will be continued (skipped). append must be here for another claims